src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs

```diff
@@ -329,6 +364,26 @@ public void ApplyForwarders(HttpContext context)
                 }
                 request.Host = HostString.FromUriComponent(currentValues.Host);
             }
+
+            if (checkPrefix && currentValues.Prefix != null)
+            {
+                // Save the original
+                requestHeaders[_options.OriginalPrefixHeaderName] =
+                    request.PathBase.HasValue ? request.PathBase.ToString() : "/";
```

Comment:
"/" and "" are not equivalent paths, especially for PathBase. PathBase is commonly empty. I understand that's awkward to  represent in this collection. It might be better to skip adding the original value when empty.

Reply:
Changed so that X-Original-Prefix is only added if PathBase has value